src/components/Contacts/ContactListTableMobile.jsx

```diff
@@ -0,0 +1,207 @@
+// React Imports
+import React, { useState } from 'react';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
+import IconButton from '@mui/material/IconButton';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import SendIcon from '@mui/icons-material/Send';
+import Typography from '@mui/material/Typography';
+import { useTheme } from '@mui/material/styles';
+// Custom Hooks Imports
+import useNotification from '@hooks/useNotification';
+// Util Imports
+import { saveToClipboard } from '@utils';
+// Component Imports
+import ContactProfileIcon from './ContactProfileIcon';
+
+/**
+ * @typedef {object} Contact
+ * @property {string} webId - The Web ID of the contact
+ * @property {string} givenName - The given name of the contact
+ * @property {string} familyName - The family name of the contact
+ */
+
+/**
+ * ContactListTableMobile - Component for displaying contacts in a DataGrid
+ *
+ * @memberof Contacts
+ * @name ContactListTableMobile
+ * @param {object} Props - The props for ContactListTableMobile
+ * @param {Contact[]} Props.contacts - The list of contacts to display
+ * @param {Function} Props.deleteContact - Function to delete a contact
+ * @param {Function} Props.handleSendMessage - Function to handle sending a message
+ * @returns {React.JSX.Element} The ContactListTableMobile component
+ */
+const ContactListTableMobile = ({ contacts, deleteContact, handleSendMessage }) => {
+  const { addNotification } = useNotification();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [openMenu, setOpenMenu] = useState(null);
+  const theme = useTheme();
+
+  const handleClick = (event, contact) => {
+    setAnchorEl(event.currentTarget);
+    setOpenMenu(contact.webId);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+    setOpenMenu(null);
+  };
+
+  const handleMenuItemClick = (action, contact) => () => {
+    action(contact);
+    handleClose();
+  };
+
+  const handleProfileClick = (contactData) => {
```

Comment:
if you change this function to the following, then clicking on the label will also load the profile:
```
  const handleProfileClick = (contact) => {
    navigate(`/contacts/${encodeURIComponent(contact.id)}`)
  };
```